server/api.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost/server/public/plugin"
+)
+
+// ServeHTTP demonstrates a plugin that handles HTTP requests by greeting the world.
+func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Request) {
+	router := mux.NewRouter()
+
+	// Middleware to require that the user is logged in
+	router.Use(p.MattermostAuthorizationRequired)
+
+	apiRouter := router.PathPrefix("/api/v1").Subrouter()
+
+	apiRouter.HandleFunc("/hello", p.HelloWorld).Methods(http.MethodGet)
+
+	router.ServeHTTP(w, r)
+}
+
+func (p *Plugin) MattermostAuthorizationRequired(next http.Handler) http.Handler {
```

Comment:
It would be awesome to hoist this into the `public` package.

Reply:
Yeah that makes sense